tests/mock_virtual_machine.h

```diff
@@ -29,14 +29,23 @@ struct MockVirtualMachine : public multipass::VirtualMachine
 {
     MockVirtualMachine(const std::string vm_name) : VirtualMachine{vm_name}
     {
-        ON_CALL(*this, current_state()).WillByDefault(Return(multipass::VirtualMachine::State::off));
+        ON_CALL(*this, start()).WillByDefault([this]() { fake_state = multipass::VirtualMachine::State::running; });
+        ON_CALL(*this, stop()).WillByDefault([this]() { fake_state = multipass::VirtualMachine::State::off; });
+        ON_CALL(*this, shutdown()).WillByDefault([this]() { fake_state = multipass::VirtualMachine::State::off; });
+        ON_CALL(*this, suspend()).WillByDefault([this]() { fake_state = multipass::VirtualMachine::State::suspended; });
+        ON_CALL(*this, current_state()).WillByDefault([this]() { return fake_state; });
```

Comment:
So you moved to mock after all :) But then you shouldn't need any of this. The advantage of the mock is that you can tell it what you want of it in the spot. For instance:
```cpp
EXPECT_CALL(mock, current_state())
    .WillOnce(Return(<desired_state>))
    .WillOnce(Return(<another_state>));
EXPECT_CALL(mock, ensure_vm_is_running()).WillRepeatedly(Throw(std::runtime_error{"Not running"}));
```

Reply:
Good to know, thanks! The problem here is that, since we are dealing with a real daemon, thus AFAICS we cannot obtain the instance object (which is the mocked VM).